ToDoApp/ToDoApp/ToDoService.swift

```diff
@@ -30,7 +30,7 @@ extension Calendar: DateChecker {}
 
 protocol ToDoService {
     func createTask(title: String, deadline: Date) async -> Result<TaskModel, TaskError>
-    func updateTask(todo: TaskModel) async -> Result<TaskModel, TaskError>
+    func updateTask(todo: TaskModel, newTitle: String, newDeadline: Date) async -> Result<TaskModel, TaskError>
```

Comment:
why do we need a new title and new deadline arguments when we are passing a todo value which should contain all the updated properties already?
btw, just noticed recently the same concern is about `updateTaskStatus` method. why do we need a separate method for updating status if there's already updateTask method?

Reply:
because when I tried todo.title = title in the createTaskViewModel so that I just need to pass only todo to updateTask function, but it gave me an error telling that todo is immutable so I can just pass the previous todo, newTitle, and newDeadline.  Therefore, I edit the method like this. But I have figured out how to fix this, and everything is fixed now.